library/src/main/java/com/datatorrent/lib/io/fs/FileSplitter.java

```diff
@@ -634,22 +634,23 @@ public boolean isDirectory()
     private String filePatternRegularExp;
 
     protected transient long lastScanMillis;
+    protected long noOfDiscoveredFilesInThisScan;
     protected transient FileSystem fs;
     protected final transient LinkedBlockingDeque<FileInfo> discoveredFiles;
     protected final transient ExecutorService scanService;
     protected final transient AtomicReference<Throwable> atomicThrowable;
 
-    private transient volatile boolean running;
+    protected transient volatile boolean running;
     protected final transient HashSet<String> ignoredFiles;
     protected transient Pattern regex;
     protected transient long sleepMillis;
 
     public TimeBasedDirectoryScanner()
     {
-      lastModifiedTimes = Maps.newHashMap();
+      lastModifiedTimes = Maps.newConcurrentMap();
```

Comment:
We discussed it on the last pull request that this wasn't the right solution and ticket was also re-opened. Then why this change